compiler/src/dotty/tools/io/ClassPath.scala

```diff
@@ -242,8 +242,8 @@ abstract class ClassPath {
       case Some((pkg, rest)) =>
         val rep = packages find (_.name == pkg) flatMap (_ findClass rest)
         rep map {
-          case x: ClassRep  => x
-          case x            => throw new FatalError("Unexpected ClassRep '%s' found searching for name '%s'".format(x, name))
+          case x: AnyClassRep => x
+          case x              => throw new FatalError("Unexpected ClassRep '%s' found searching for name '%s'".format(x, name))
```

Comment:
I'm not saying that the check was actually performed, I'm saying that the code was written assuming that the check would be performed, which apparently it wasn't.